components/Tabs/Tabs.js

```diff
@@ -1,61 +1,65 @@
 class TabLink {
-  constructor(tabElement){
+  constructor(tabElement) {
     // assign this.tabElement to the tabElement DOM reference
-    // this.tabElement;
-    
+    this.tabElement = tabElement;
+
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
-    
+    this.tabData = this.tabElement.dataset.tab;
+
+
+
     // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
-    
-    /* <- Delete this comment block when you work on the if statement
+
+
     // Check to see if this.tabData is equal to 'all'
-    if(){
+    if (this.tabData === "all") {
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+      this.cards = document.querySelectorAll(".card");
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
     }
-    /* <- Delete this comment block when you work on the if statement
 
-     // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
-    // this.cards = Array.from(this.cards).map();
+
+    // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
+
+    this.cards = Array.from(this.cards).map(card => new TabCard(card));
 
     // Add a click event that invokes this.selectTab
-    // this.tabElement.addEventListener();
+    this.tabElement.addEventListener('click', () => this.selectTab());
   }
 
-  selectTab(){
+  selectTab() {
 
     // Select all elements with the .tab class on them
-    // const tabs = document.querySelectorAll();
-    
-    // Iterate through the NodeList removing the .active-tab class from each element
-    // tabs.forEach()
+    const tabs = document.querySelectorAll(".tab");
 
+    // Iterate through the NodeList removing the .active-tab class from each element
+    // OR should I try tabs.forEach(classList.remove(".active-tab"));
+    tabs.forEach(tab => tab.classList.remove(".active-tab"));
+    // tabs.forEach(tab.classList.remove('active-tab'));
     // Select all of the elements with the .card class on them
-    // const cards = ;
+    const cards = document.querySelectorAll(".card");
 
     // Iterate through the NodeList setting the display style each one to 'none'
-    // cards.forEach()
-    
+    cards.forEach(card => card.style = "display:none");
+
```

Comment:
I think this works, but it would be better if you did `card.style.display = 'none'`